codalab/worker/docker_image_manager.py

```diff
@@ -151,20 +176,20 @@ def image_availability_state(image_spec, success_message, failure_message):
                 image = self._docker.images.get(image_spec)
                 digests = image.attrs.get('RepoDigests', [image_spec])
                 digest = digests[0] if len(digests) > 0 else None
-                with self._lock:
-                    self._image_cache[digest] = ImageCacheEntry(
-                        id=image.id,
-                        digest=digest,
-                        last_used=time.time(),
-                        virtual_size=image.attrs['VirtualSize'],
-                        marginal_size=image.attrs['Size'],
-                    )
+                new_timestamp = str(time.time())
+                image.tag(self.CACHE_TAG, tag=new_timestamp)
+                image.reload()
```

Comment:
@bkgoksel I also added a `reload()` function after tagging an image. Sometimes I noticed that the new tag doesn't show up automatically. Reloading the image can help resolve this issue.